airflow/models/param.py

```diff
@@ -14,340 +14,9 @@
-
-        with DAG(...) as dag:
-          EmailOperator(subject=dag.param('subject', 'Hi from Airflow!'))
-
-    :param current_dag: Dag being used for parameter.
-    :param name: key value which is used to set the parameter
-    :param default: Default value used if no parameter was set.
-    """
-
-    def __init__(self, current_dag: DAG, name: str, default: Any = NOTSET):
-        if default is not NOTSET:
-            current_dag.params[name] = default
-        self._name = name
-        self._default = default
-        self.current_dag = current_dag
-
-    def iter_references(self) -> Iterable[tuple[Operator, str]]:
-        return ()
-
-    def resolve(self, context: Context, *, include_xcom: bool = True) -> Any:
-        """Pull DagParam value from DagRun context. This method is run during ``op.execute()``."""
-        with contextlib.suppress(KeyError):
-            if context["dag_run"].conf:
-                return context["dag_run"].conf[self._name]
-        if self._default is not NOTSET:
-            return self._default
-        with contextlib.suppress(KeyError):
-            return context["params"][self._name]
-        raise AirflowException(f"No value could be resolved for parameter {self._name}")
-
-    def serialize(self) -> dict:
-        """Serialize the DagParam object into a dictionary."""
-        return {
-            "dag_id": self.current_dag.dag_id,
-            "name": self._name,
-            "default": self._default,
-        }
-
-    @classmethod
-    def deserialize(cls, data: dict, dags: dict) -> DagParam:
-        """
-        Deserializes the dictionary back into a DagParam object.
-
-        :param data: The serialized representation of the DagParam.
-        :param dags: A dictionary of available DAGs to look up the DAG.
-        """
-        dag_id = data["dag_id"]
-        # Retrieve the current DAG from the provided DAGs dictionary
-        current_dag = dags.get(dag_id)
-        if not current_dag:
-            raise ValueError(f"DAG with id {dag_id} not found.")
-
-        return cls(current_dag=current_dag, name=data["name"], default=data["default"])
-
-
-def process_params(
-    dag: DAG,
-    task: Operator,
-    dagrun_conf: dict[str, Any] | None,
-    *,
-    suppress_exception: bool,
-) -> dict[str, Any]:
-    """Merge, validate params, and convert them into a simple dict."""
-    from airflow.configuration import conf
-
-    dagrun_conf = dagrun_conf or {}
+from __future__ import annotations
 
-    params = ParamsDict(suppress_exception=suppress_exception)
-    with contextlib.suppress(AttributeError):
-        params.update(dag.params)
-    if task.params:
-        params.update(task.params)
-    if conf.getboolean("core", "dag_run_conf_overrides_params") and dagrun_conf:
-        logger.debug("Updating task params (%s) with DagRun.conf (%s)", params, dagrun_conf)
-        params.update(dagrun_conf)
-    return params.validate()
+from airflow.sdk.definitions.param import *  # noqa: F403
```

Comment:
Do we need `*`, or could we do
```suggestion
from airflow.sdk.definitions.param import Param, ParamsDict

__all__ = ["Param", "ParamsDict"]
```

Reply:
Good call, we do not need all. This will do